inc/datahandlers/user.php

```diff
@@ -888,6 +888,20 @@ function verify_checkfields()
 	}
 
 	/**
+	 * Verifies if the selected timezone is valid.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_timezone()
+	{
+		$user = &$this->data;
+
+		$timezones = get_supported_timezones();
+
+		return array_key_exists($user['timezone'], $timezones);
```

Comment:
@psde
- `verify_referrer` does set an error if the user inserted a value.
- `verify_lastvisit` value is not an user input.

@Destroy666x Then move this inside `verify_options`?

I believe something around the lines of...

```PHP
function verify_timezone()
{
	$user = &$this->data;

	$timezones = get_supported_timezones();

	if(!array_key_exists($user['timezone'], $timezones))
	{
 		$this->set_error("invalid_timezone");
 		return false;
		// or force the timezone here
		//$user['timezone'] = $mybb->settings['timezoneoffset'];
	}

	return true;
}

if($this->method == "insert" && array_key_exists('timezone', $user))
{
	$this->verify_timezone();
}
```

Should be.

@euantorano